### PR TITLE
Filter, Search  and Pagination

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights
Related [document](https://tabdullai.notion.site/ME-PAGINATION-SEARCH-AND-FILTER-469085b3ac984435980baa0ffa56d8f2)
API [PR 623](https://github.com/massenergize/api/pull/623)

### CODE REVIEW

**Comment by frimpongopoku** [comment]
@abdullai-t, please add a DEMO of how this thing works.

File: app/app.js [hunk 1]
```diff
@@ -48,10 +48,10 @@ const SENTRY_DSN =
     ? process.env.REACT_APP_SENTRY_PROD_DSN
     : process.env.REACT_APP_SENTRY_DEV_DSN;
 
-Sentry.init({
-  dsn: SENTRY_DSN,
-  integrations: [new BrowserTracing()],
-});
+// Sentry.init({
```
Comment (BradHN1) [inline app/app.js:51]: Is Sentry not working?  Or do we not want to call this?

Comment (abdullai-t) [reply app/app.js:51]: I turned it off to prevent unnecessary alerts during development. nice catch 👍🏾

**Comment by abdullai-t** [comment]
## Demo Video

https://user-images.githubusercontent.com/42780120/220203573-ec5d19d7-f37c-4745-9b12-509e2bf05608.mp4

File: app/containers/MassEnergizeSuperAdmin/Actions/CreateNewActionForm.js [hunk 2]
```diff
@@ -46,8 +46,7 @@ class CreateNewActionForm extends Component {
   static getDerivedStateFromProps(props, state) {
     const { communities, tags, vendors, ccActions, auth } = props;
     const fullyMountedNeverRunThisAgain =
-      communities &&
-      communities.length &&
+      communities && communities.length;
```
Comment (BradHN1) [inline app/containers/MassEnergizeSuperAdmin/Actions/CreateNewActionForm.js:49]: shouldn't have have the semi-colon

File: app/containers/MassEnergizeSuperAdmin/Summary/CommunitySwitch.js [hunk 3]
```diff
@@ -33,7 +33,8 @@ class CommunitySwitch extends PureComponent {
 
   getDropdownCommunities() {
     var { auth, communities } = this.props;
-    communities = communities || [];
+
+    communities = communities && communities|| [];
```
Comment (BradHN1) [inline app/containers/MassEnergizeSuperAdmin/Summary/CommunitySwitch.js:37]: this doesn't look correct

File: app/containers/MassEnergizeSuperAdmin/Testimonials/CreateNewTestimonialForm.js [hunk 4]
```diff
@@ -58,18 +57,18 @@ class CreateNewTestimonialForm extends Component {
...
 
-    const coms = communities.map((c) => ({
+    const coms = (communities ||[]).map((c) => ({
       ...c,
       id: "" + c.id,
       displayName: c.name,
     }));
 
-    const vends = vendors.map((c) => ({
+    const vends = (vendors && vendors ||[]).map((c) => ({
```
Comment (BradHN1) [inline app/containers/MassEnergizeSuperAdmin/Testimonials/CreateNewTestimonialForm.js:66]: doesn't look correct

**Review by BradHN1 — APPROVED** [review]
LGTM - w couple minor fixes

**Pull request merged**